### PR TITLE
FeatureB, miss component DropDownTraveller

### PR DESCRIPTION
**Pull request by lxndrMr**

### CODE REVIEW

File: client/src/App.jsx [hunk 1]
```diff
@@ -1,11 +1,99 @@
...
+        galactapediaData={galactapediaData}
+      />
+      {reservationData && (
+        <section>
+          <h2>Récapitulatif de la réservation :</h2>
+          <p>Départ de : {reservationData.selectedDeparture}</p>
+          <p>Arrivée à : {reservationData.selectedArrival}</p>
+          <p>Date de départ : {reservationData.selectedArrival}</p>
+          <p>Type de voyage : {reservationData.selectedTripType}</p>
+          {/* <p>Nombre de passagers : {reservationData.selectedTravellerCount}</p> */}
```
Comment (FabienRollet) [inline client/src/App.jsx]: l87 to l92 -> needed without other components, to be deleted after.

File: client/src/components/ReservationModule.jsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
+import { useState } from "react";
+import PropTypes from "prop-types";
+import Arrival from "./Arrival";
+import DateSelect from "./DateSelect";
+import Departure from "./Departure";
+import DropdownRoundTrip from "./DropdownRounTrip";
+// import DropDownTraveller from "./DropDownTraveller";
```
Comment (FabienRollet) [inline client/src/components/ReservationModule.jsx]: Waiting for PR approuval of this componant

File: client/src/components/ReservationModule.jsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    
+    if (selectedDeparture && selectedArrival && selectedDate && selectedTripType /* && selectedTravellerCount */) {
+      handleReservationSubmit(formData);
+    } else {
+      alert("Veuillez remplir tous les champs du formulaire.");
+    }
+  };
+
+  return (
+    <form onSubmit={handleSubmit}>
```
Comment (FabienRollet) [inline client/src/components/ReservationModule.jsx:27]: fieldset tag + legend tag required

File: client/src/components/Arrival.jsx [hunk 4]
```diff
@@ -1,20 +1,42 @@
...
+    setInputValue(selectedValue);
+    handleArrivalSelect(selectedValue);
   };
+
   return (
-    <>
+    <div>
       <label htmlFor="arrival">Arrivée à :</label>
-      <input
+      <select
```
Comment (FabienRollet) [inline client/src/components/Arrival.jsx]: datalist tag would be better here

File: client/src/components/Departure.jsx [hunk 5]
```diff
@@ -1,22 +1,42 @@
...
-            onChange={handleDeparture}
-            />
-        </>
-    )
-}
\ No newline at end of file
+  return (
+    <div>
+      <label htmlFor="departure">Départ de :</label>
+      <select
```
Comment (FabienRollet) [inline client/src/components/Departure.jsx]: datalist tag would be better here

File: client/src/App.jsx [hunk 6]
```diff
@@ -1,11 +1,99 @@
-import DropdownRoundTrip from "./components/DropdownRounTrip";
+import { useState } from "react";
+import ReservationForm from "./components/ReservationModule";
```
Comment (Yoann-Fortin) [inline client/src/App.jsx]: Fix typo please

File: client/src/components/Arrival.jsx [hunk 7]
```diff
@@ -1,20 +1,42 @@
...
+import PropTypes from "prop-types";
 
-export default function Arrival() {
-  const [selectedArrival, setSelectedArrival] = useState("");
+export default function Arrival({ handleArrivalSelect, suggestions }) {
+  const [inputValue, setInputValue] = useState("");
 
-  const handleArrival = (e) => {
-    setSelectedArrival(e.target.value);
+  const handleSelectSuggestion = (event) => {
```
Comment (Yoann-Fortin) [inline client/src/components/Arrival.jsx]: Dead code here

File: client/src/components/Arrival.jsx [hunk 8]
```diff
@@ -1,20 +1,42 @@
...
+    const selectedValue = event.target.value;
+    setInputValue(selectedValue);
+    handleArrivalSelect(selectedValue);
   };
+
   return (
-    <>
+    <div>
```
Comment (Yoann-Fortin) [inline client/src/components/Arrival.jsx]: Why this div ?

File: client/src/components/Arrival.jsx [hunk 9]
```diff
@@ -1,20 +1,42 @@
...
         id="arrival"
-        type="text"
-        value={selectedArrival}
-        onChange={handleArrival}
-      />
-    </>
+        value={inputValue}
+        onChange={handleSelectSuggestion}
+      >
+        <option value="" disabled hidden>
```
Comment (Yoann-Fortin) [inline client/src/components/Arrival.jsx]: Can you explain me this line ?

File: client/src/components/DateSelect.jsx [hunk 10]
```diff
@@ -1,19 +1,29 @@
...
-            <input
-            id="departureDate"
-            type="date"
-            value={departureDate}
-            onChange={handleDateChange}
-            />
-        </>
-    )}
+export default function DateSelect({ handleDateSelect }) {
+  const [departureDate, setDepartureDate] = useState("");
```
Comment (Yoann-Fortin) [inline client/src/components/DateSelect.jsx]: Dead code

File: client/src/components/DateSelect.jsx [hunk 11]
```diff
@@ -1,19 +1,29 @@
...
+
+  const handleDateChange = (event) => {
+    const selectedDate = event.target.value;
```
Comment (Yoann-Fortin) [inline client/src/components/DateSelect.jsx]: And here

File: client/src/components/DateSelect.jsx [hunk 12]
```diff
@@ -1,19 +1,29 @@
...
+    setDepartureDate(selectedDate);
+    handleDateSelect(selectedDate);
+  };
+
+  return (
+    <div>
```
Comment (Yoann-Fortin) [inline client/src/components/DateSelect.jsx]: Why this tag ?

File: client/src/components/Departure.jsx
Comment (Yoann-Fortin) [inline client/src/components/Departure.jsx:1]: Same component but repeat ?

**Review by Yoann-Fortin — CHANGES_REQUESTED** [review]

File: client/src/components/Arrival.jsx (hunk 9 above)
Comment (lxndrMr) [reply client/src/components/Arrival.jsx]: To have an Default Value but unusable by the user

File: client/src/components/DestinationSelect.jsx [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+  ListBox,
+  ListBoxItem,
+  Popover,
+} from "react-aria-components";
+
+function DestinationSelect({ label, handleSelect, suggestions }) {
+  return (
+    <ComboBox>
+      <Label>{label}</Label>
+      <div>
```
Comment (Yoann-Fortin) [inline client/src/components/DestinationSelect.jsx]: Why use a div tag here ?

File: client/src/components/ReservationModule.jsx [hunk 14]
```diff
@@ -0,0 +1,65 @@
+import PropTypes from "prop-types";
+import { useState } from "react";
+import DateSelect from "./DateSelect";
+import DropdownRoundTrip from "./DropdownRounTrip";
+import DestinationSelect from "./DestinationSelect";
+// import DropDownTraveller from "./DropDownTraveller";
```
Comment (Yoann-Fortin) [inline client/src/components/ReservationModule.jsx]: commented code is prohibited

File: client/src/components/ReservationModule.jsx [hunk 15]
```diff
@@ -0,0 +1,65 @@
...
+        label="Arrivée à"
+        value={formData.selectedArrival}
+        handleSelect={(value) => handleInputChange("selectedArrival", value)}
+        suggestions={galactapediaData.data}
+      />
+      <DateSelect
+        handleSelect={(value) => handleInputChange("selectedDate", value)}
+        value={formData.selectedDate}
+      />
+      {/* <DropDownTraveller onChange={(count) => handleInputChange('selectedTravellerCount', count)} /> */}
```
Comment (Yoann-Fortin) [inline client/src/components/ReservationModule.jsx]: same here

**Review by Yoann-Fortin — CHANGES_REQUESTED** [review]

**Review by Yoann-Fortin — APPROVED** [review]

**Review by Yoann-Fortin — APPROVED** [review]

**Pull request merged**